### DISCUSSION TITLE
NFT create listing error on opensea testnet

### DISCUSSION THREAD

**Discussion by Gregusko** [ProjectOpenSea/seaport-js#415] [General]
Trying to create listing code on testnet.opensea. 

Have ERC721 smart contract published to Goerli network and have minted 2 NFTS sucessfully. NFT's are in IPFS storage with json metadata and all can be seen and accessed via opeasea SDK. https://docs.opensea.io/reference/get_nft

Have Ledger device attached to metamask wallet. 

Running this code, get error: Points to issue with Ledger, but wallet address and signer return expected ID. 

**Failed to create sell order for token ID 1: Error: Specified accountAddress is not available through wallet or provider. Accounts available: none**

async function main() {
    try {
        const signer = await getLedgerWallet();
        const address = await signer.getAddress();
        console.log("Address from signer:", address); 
  
        const seaport = new OpenSea.OpenSeaPort(signer, {
            networkName: NETWORK === "GOERLI" ? "goerli" : "ethereum"
        }, (msg) => {
            console.log("[OpenSea SDK]", msg);
        });
        
        const tokenAddress = process.env.CONTRACT_ADDRESS;
        const accountAddress = process.env.WALLET_ADDRESS;

        for (let i = 1; i <= totalNFTs; i++) {
            console.log(`Checking Token ID ${i}...`);
        
            const existsOnOpenSea = await checkIfAssetExistsOnOpenSea(i);
            console.log(`Does Token ID ${i} exist on OpenSea?`, existsOnOpenSea);
        
            if (!existsOnOpenSea) {
                const price = rarityPricing.calculatePriceForTokenId(i);
                console.log(`Calculated price for Token ID ${i}:`, price);
        
                try {
                    console.log(`Attempting to create sell order for Token ID ${i}...`);
                    
                    const tokenId = i.toString();
                    const sellOrder = await seaport.createListing({
                        asset: {
                            tokenAddress,
                            tokenId,
                        },
                        accountAddress,
                        startAmount: price,
                    });
                    console.log(`Created sell order for token ID ${i}`);
                } catch (error) {
                    console.error(`Failed to create sell order for token ID ${i}:`, error);
                }
            } else {
                console.log(`Token ID ${i} already exists on OpenSea, skipping...`);
            }
        }
    } catch (error) {
        console.error("Error in main function:", error);
    }
  }

**Comment by ryanio** [comment]
Did you first call eth_requestAccounts?

**Reply by Gregusko** [reply]
I didn't try that. I think I'm accessing the endpoint wrong anyhow. I'm trying to use seaport to create a listing on opensea before minting so I can evoke lazy minting, and only mint at time of purchase
